client/lib/signup/flow-controller.js

```diff
@@ -193,8 +194,9 @@ assign( SignupFlowController.prototype, {
 
 	_process: function() {
 		const currentSteps = this._flow.steps;
-		const signupProgress = filter( getSignupProgress( this._reduxStore.getState() ), step =>
-			includes( currentSteps, step.stepName )
+		const signupProgress = getFilteredSteps(
+			this._flowName,
+			getSignupProgress( this._reduxStore.getState() )
```

Comment:
I tested this branch rebased on master.

Using `getFilteredSteps` returns filtered steps **without** those steps that have been excluded.

The problem is that it won't grab excluded steps whose states are pending, such as `'plans-business'` in the `business` flow, and hence won't assign them to `pendingSteps` [below](**https://github.com/Automattic/wp-calypso/pull/35139/files#diff-9b8826c512c150c2396692c766af5c35R201**)

The pending steps are therefore never processed.

For the business flow, the pending `'plans-business'` step is processed once a site is created and we have a site slug. 

Since `'plans-business'` is filtered out however, the api request function (`addPlanToCart()`) is never called, and we never reach the checkout in the `business` flow. It hangs on the site creation page:

<img width="400" alt="Screen Shot 2019-08-07 at 12 39 14 pm" src="https://user-images.githubusercontent.com/6458278/62593641-e09a2f00-b91a-11e9-876c-0b2385af7c2c.png">

I know this PR is not complete, but just flagging it so we can consider how we might fix our flow forking woes using the new progress data model.